consensus/src/simplex/actors/voter/actor.rs

```diff
@@ -239,25 +322,23 @@ impl<
     }
 
     /// Handle a timeout.
-    async fn timeout<Sp: Sender, Sr: Sender>(
+    async fn handle_timeout<Sp: Sender, Sr: Sender>(
         &mut self,
         batcher: &mut batcher::Mailbox<S, D>,
         pending_sender: &mut WrappedSender<Sp, Voter<S, D>>,
         recovered_sender: &mut WrappedSender<Sr, Voter<S, D>>,
     ) {
         // Set timeout fired
         let current_view = self.state.current_view();
-        let was_retry = self
-            .state
-            .handle_timeout(current_view, self.context.current());
+        let (was_retry, Some(nullify)) = self.state.handle_timeout(current_view) else {
+            return;
+        };
 
         // When retrying we immediately re-share the certificate that let us enter
         // this view so slow peers do not stall our next round.
-        let past_view = current_view - 1;
         if was_retry
-            && past_view > 0
             && !self
-                .rebroadcast_entry_certificates(recovered_sender, past_view)
+                .rebroadcast_entry_certificates(recovered_sender, current_view)
                 .await
         {
             warn!(
```

Comment:
I think it would simplify the code flow to simply do this `warn` inside of `rebroadcast_entry_certificates`. To keep the behavior the same, don't print the warning unless you get to the bottom `false` of the function. That is, don't print it if the current view is <= 1